piet/src/error.rs

```diff
@@ -4,35 +4,28 @@ use std::fmt;
 
 /// An error that can occur while rendering 2D graphics.
 #[derive(Debug)]
-pub struct Error(Box<ErrorKind>);
-
-#[derive(Debug)]
-pub enum ErrorKind {
+#[non_exhaustive]
```

Comment:
The thinking for this in general is to allow us to add new variants without it being an API breaking change, requiring a major version bump. This isn't super relevant right now, since we're not even at 0.1, but I do think it's often worth once things mature a bit.

In any case I hadn't thought about this much, it was really just updating the existing code to use the attribute instead of a hidden `__NonExhaustive` type.

I could imagine matching on the error being useful, so I see your point, but my inclination is to just leave things as-is for now?

Reply:
Well my thinking is that the only people for whom adding new variants would be a breaking change are people who already match against every single variant. Now if they want to match every single variant, then adding new variants hidden by `non_exhaustive` means that we can "break" their code by making it no longer complete.

Thus I view this mostly as a trade-off between breaking compilation and breaking intended behavior. Personally I would much prefer the compilation error so that I could handle the new variants, or add a catch-all if I don't care about them and thus also protect myself from seeing future compilation breakage.

In practice I don't really match on all error variants anywhere so either way would be fine. However in theory if I would, I would prefer the compiler errors for runtime correctness.

Reply:
if I were arguing (I'm not arguing!) I would say that by marking this as `non_exhaustive`, we are explicitly indicating that it is incorrect to assume you can handle all cases individually, and that you must explicitly handle the "some other error" case.

My inclination is to just roll up this patch and we can revisit this if and when we get more rigorous about our error handling, generally.

Reply:
Sure, I'm still gathering experience with Rust's error handling anyway. I'll have a better view later down the line.